# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1239060781 14400
#      Mon Apr 06 19:33:01 2009 -0400
# Node ID 3a5d41b382abf2e63736f909923f253922f4de56
# Parent  5ecdb3d3568fe46a4720392d132371cb961e0fc2
__gnuplot_get_var__.m: Correct misleading error messages, and remove dangling/commented code.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,12 @@
 2009-04-06  Ben Abbott <bpabbott@mac.com>
 
+	* plot/__gnuplot_get_var__.m: Correct misleading error messages, and
+	remove dangling/commented code.
 	* plot/__gnuplot_default_font__.m: Remove file from archive.
 	* plot/__go_draw_axes__.m, plot/__go_draw_figure__.m: Allow
 	fontsize to be specified for all terminals with fontname == "*".
 
 2009-04-06  Jaroslav Hajek <highegg@gmail.com>
 
 	* polynomial/polyscale.m: New function.
 	* polynomial/polytrans.m: New function.
diff --git a/scripts/plot/__gnuplot_get_var__.m b/scripts/plot/__gnuplot_get_var__.m
--- a/scripts/plot/__gnuplot_get_var__.m
+++ b/scripts/plot/__gnuplot_get_var__.m
@@ -38,17 +38,17 @@ function gp_var_value = __gnuplot_get_va
 
   if (numel (h) == 1 && isfigure (h))
     ostream = get (h, "__plot_stream__");
   else
     ostream = h;
   endif
   if (numel (ostream) < 1)
     error ("__gnuplot_get_var__: stream to gnuplot not open");
-  elseif (ispc ()) # || true
+  elseif (ispc ())
     if (numel (ostream) == 1)
       error ("__gnuplot_get_var__: Need mkfifo that is not implemented under Windows");
     endif
     use_mkfifo = false;
     istream = ostream(2);
     ostream = ostream(1);
   else
     use_mkfifo = true;
@@ -57,17 +57,17 @@ function gp_var_value = __gnuplot_get_va
 
   if (use_mkfifo)
     gpin_name = tmpnam ();
 
     ## Mode: 6*8*8 ==  0600
     [err, msg] = mkfifo (gpin_name, 6*8*8);
 
     if (err != 0)
-      error ("__gnuplot_get_var__: Can not open fifo (%s)", msg);
+      error ("__gnuplot_get_var__: Can not make fifo (%s)", msg);
     endif
   endif
 
   gp_var_name = strtrim (gp_var_name);
   n = min (strfind (gp_var_name, " "), strfind (gp_var_name, ",")) - 1;
   if (isempty (n))
     n = numel (gp_var_name);
   endif
@@ -77,17 +77,17 @@ function gp_var_value = __gnuplot_get_va
     ## Notes: Variables may be undefined if user closes gnuplot by "q"
     ## or Alt-F4. Further, this abrupt close also requires the leading
     ## "\n" on the next line.
     if (use_mkfifo)
       fprintf (ostream, "\nset print \"%s\";\n", gpin_name);
       fflush (ostream);
       [gpin, err] = fopen (gpin_name, "r");
       if (err != 0)
-        error ("__gnuplot_get_var__: Can not open fifo (%s)", msg);
+        error ("__gnuplot_get_var__: Can not open fifo.");
       endif
       gp_cmd = sprintf ("\nif (exists(\"%s\")) print %s; else print NaN\n",
                         gp_var_name(1:n), gp_var_name);
       fputs (ostream, gp_cmd);
 
       ## Close output file, to force it to be flushed
       fputs (ostream, "set print;\n");
       fflush (ostream);
